### PR TITLE
fix: clean up outdated dependencies & modules and fix tests

### PR DESCRIPTION
**Pull request by Sevenannn**
## 🗣 Description

Fix the datafusion federation fork to pass all test. Aligning to the upstream latest changes and removed outdated examples (postgres-partial, sqlite-partial, sqlite), removed outdated mod cconnectorx, updated unit tests to follow upstream changes. Also did some formatting changes to pass the clippy/ format tests.

### CODE REVIEW

File: examples/examples/flight-sql.rs [hunk 1]
```diff
@@ -39,14 +40,20 @@ async fn main() -> Result<()> {
     sleep(Duration::from_secs(3)).await;
 
     // Local context
-    let state = SessionContext::new().state();
+    // let state = SessionContext::new().state();
```
Comment (sgrebnov) [inline examples/examples/flight-sql.rs:43]:
```suggestion
```

File: examples/examples/flight-sql.rs [hunk 2]
```diff
@@ -39,14 +40,20 @@ async fn main() -> Result<()> {
...
     let known_tables: Vec<String> = ["test"].iter().map(|&x| x.into()).collect();
 
     // Register FederationAnalyzer
     // TODO: Interaction with other analyzers & optimizers.
-    let state = state
-        .add_analyzer_rule(Arc::new(FederationAnalyzerRule::new()))
-        .with_query_planner(Arc::new(FederatedQueryPlanner::new()));
+    // let mut state = state
```
Comment (sgrebnov) [inline examples/examples/flight-sql.rs:48]: Should we remove this completely?

Comment (Sevenannn) [reply examples/examples/flight-sql.rs:48]: Yes, let me remove it

File: sources/sql/src/lib.rs [hunk 3]
```diff
@@ -1008,10 +1004,10 @@ mod tests {
                 "SELECT COUNT(CASE WHEN a > 0 THEN a ELSE 0 END) FROM app_table",
                 r#"SELECT count(CASE WHEN (remote_table.a > 0) THEN remote_table.a ELSE 0 END) FROM remote_table"#,
             ),
-            // different tables in single aggregation expression
+            // // different tables in single aggregation expression
             (
-                "SELECT COUNT(CASE WHEN app_table.a > 0 THEN app_table.a ELSE foo.df_table.a END) FROM app_table, foo.df_table",
-                r#"SELECT count(CASE WHEN (remote_table.a > 0) THEN remote_table.a ELSE remote_table.a END) FROM remote_table JOIN remote_table ON true"#,
+                "SELECT COUNT(CASE WHEN appt.a > 0 THEN appt.a ELSE dft.a END) FROM app_table as appt, foo.df_table as dft",
```
Comment (sgrebnov) [inline sources/sql/src/lib.rs:1009]: Not sure how the test works as original and rewritten table names are the same - `appt` - is it expected?

File: sources/sql/src/lib.rs [hunk 4]
```diff
@@ -1008,10 +1004,10 @@ mod tests {
                 "SELECT COUNT(CASE WHEN a > 0 THEN a ELSE 0 END) FROM app_table",
                 r#"SELECT count(CASE WHEN (remote_table.a > 0) THEN remote_table.a ELSE 0 END) FROM remote_table"#,
             ),
-            // different tables in single aggregation expression
+            // // different tables in single aggregation expression
```
Comment (sgrebnov) [inline sources/sql/src/lib.rs:1007]:
```suggestion
```

**Review by sgrebnov — APPROVED** [review]

**Pull request merged**